editoast/src/views/v2/train_schedule.rs

```diff
@@ -255,8 +268,18 @@ async fn put(
     Ok(Json(ts_result.into()))
 }
 
-#[derive(Deserialize, Serialize, Clone, Debug, ToSchema)]
-struct SimulationOutput {
+// TODO: When can the simulation fail ?
+#[derive(Serialize, Deserialize, Clone, Debug, ToSchema)]
+#[serde(tag = "status", rename_all = "snake_case")]
+enum SimulationResult {
+    Success(Box<SimulationResultSuccess>),
```

Comment:
Why do we need to `Box` here? Is it because the enum's variants have very different size and `clippy` is complaining about it ([like this](https://rust-lang.github.io/rust-clippy/master/index.html#/large_enum_variant))? If I understand correctly, the `PathFindingFailed` variant is for an error case, so maybe we don't care too much to run in a suboptimal way (a lot of memory wasted in this case) for an error case? Maybe a `#[allow(clippy::large_enum_variant)]` would be ok here? I'm really exploring the idea here, I'm not sure myself, happy to take feedback on this idea.